linkml-schema/reproschema.yaml

```diff
@@ -0,0 +1,669 @@
+    is_a: Agent
+    slots:
+      - subject_id
+    class_uri: reproschema:Participant
+  Protocol: # TODO multiple types
+    title: Protocol
+    description: A representation of a study which comprises one or more assessments.
+    is_a: CreativeWork
+    slots:
+    - about
+    - altLabel
+    - associatedMedia
+    - compute
+    - cronTable
+    - description
+    - landingPage
+    - messages
+    - prefLabel
+    - schemaVersion
+    - ui
+    - version
+    slot_usage:
+      ui:
+        range: UI
+    class_uri: reproschema:Protocol
+  Response: # multiple types
+    title: Response
+    description: Describes the response of an item.
+    is_a: CreativeWork
+    slots:
+    - isAbout
+    - value #TODO it was schema:value
+    - wasAttributedTo
+    slot_usage:
+      value:
+        any_of:
+          - range: MissingType
+          - range: decimal
+          - range: boolean
+          - range: string
+          - range: uri
+          - range: StructuredValue
+    class_uri: reproschema:Response
+  ResponseActivity: # multiple types
+    title: ResponseActivity
+    description:
+      Captures information about some action that took place. It also links to information
+      (entities) that were used during the activity
+    is_a: CreativeWork
+    slots:
+      - endedAtTime
+      - generated
+      - inLanguage
+      - startedAtTime
+      - used
+      - wasAssociatedWith
+    class_uri: reproschema:ResponseActivity
+  ResponseOption:
+    title: Response option
+    description: An element (object or by URL)to describe the properties of response of the Item.
+    slots:
+    - choices
+    - datumType
+    - maxValue
+    - minValue
+    - multipleChoice
+    - unitOptions
+    - valueType
+    class_uri: reproschema:ResponseOption
+  SoftwareAgent: # TODO multiple types
+    title: Software Agent
+    description:
+      Captures information about some action that took place. It also links to information
+      (entities) that were used during the activity
+    slots:
+    - version
+    - url
+    class_uri: reproschema:SoftwareAgent
+  StructuredValue:
+    is_a: CreativeWork
+    class_uri: schema:StructuredValue
+  Thing: # todo should we switch
+    class_uri: schema:Thing
+  UI:
+    title: todo
+    description:
+    - todo
+    slots:
+      - order
+      - addProperties
+      - overrideProperties
+      - shuffle
+      - allow
+      - inputType
+      - readonlyValue
+    class_uri: reproschema:UI
+  UnitOption:
+    title: Unit options
+    description: An object to represent a human displayable name alongside the more formal value for units.
+    slots:
+    - prefLabel
+    - value
+    slot_usage:
+      value:
+        any_of:
+          - range: uri
+          - range: langString
+    class_uri: reproschema:UnitOption
+  VideoObject:
+    title: Video Object
+    is_a: MediaObject
+    class_uri: schema:VideoObject
+
+enums:
+  AllowedType:
```

Comment:
I think this list should also contain `Skipped` or the `activity1.jsonld` form the examples should be changed

Reply:
this one should really be `AllowSkip` and the other one in missing enum `Skipped`. `activity1` should be adjusted for the new flag. 

btw, for any of the enums, the autogenerated python needs to be updated to have `reproschema:` in front of the values.

Reply:
I'm not sure that I understand, so let me confirm:
- in `AllowedType` enum I should add one more permissible value: `AllowSkip`
- `MissingType` should leave as it is with `Skipped`
- in the `activity1` I should use `AllowSkip`

Reply:
I also noticed that in the pydantic you're changing `lang String` to `str`, so perhaps I should move `lang String` to `types` as in the example I posted yesterday. However we will loose the check from `shacl`

Reply:
confirming the first question.

pydantic was auto generated. the only thing i changed were the enums. the class `langString` in the model now has slots similar to `prefix` in the linkml model as chris had suggested. so technically it should be a class in pydantic with two fields. can you give an example of where you are seeing the change?

Reply:
they must have changed something very recently... 
when I updated linkml indeed I see difference only in `enums`.

However I'm not sure if I understand why `prefLabel` has `Optional[Dict[str, str]]` and `value` has `langString`

Reply:
pushed a few changes for `slot_usage`.  however, it looks like whenever range is `langString` it gets replaced with `Dict[str, str]`. this could be a bug. you may want to check with the linkml folks. if `langString` is in a any of range it stays as `langString`.

however
```yaml
any_of:
- range: langString
```

turns into a weird `Dict[str, LangString]` which is definitely incorrect. so likely a bug in linkml. although i don't think union of a single type: `Union[LangString]` makes sense from a typing point of view.

Reply:
yes, it's very inconsistent..

but actually having type `Dict[str, str]` makes this version valid with the data. If you use really `langString` it will fail since the keys do not match

Reply:
maybe we will simply create a class that we list all possible languages we use as attributes, so you can get in pydantic something like this:
```
class LangString(ConfiguredBaseModel):
    en: str = Field(..., description="""The english version of the string""")
    es: str = Field(..., description="""The spanish version of the sting.""")
```

Reply:
i understand we can also set a handful of languages in yaml and add `extra=Extra.allow` to the pydantic version

Reply:
btw. I think you had `Dict[str, str]` instead of `langString` because you used `key: true`. I haven't used it before, but  in a way it makes sense that it translated to `Dcit[str, str]`. Still should be consistent though